lib/sinks/gcs.js

```diff
@@ -69,8 +89,15 @@ class SinkGCS {
     }
 
     exist(filePath) {
-        const src = this._bucket.file(filePath);
         return new Promise((resolve, reject) => {
+            const pathname = path.join(this._rootPath, filePath);
+
+            if (pathname.indexOf(this._rootPath) !== 0) {
+                return reject(new Error(`Directory traversal - ${filePath}`));
+            }
+
+            const src = this._bucket.file(pathname);
+
             src.exists(error => {
                 if (error) return reject(error);
                 return resolve();
```

Comment:
this is the same behaviour as for fs sink exists method? Ie. Resolves if the file exists, rejects for anything else?

Reply:
Yeah. We need to make them have the same behaviour. The difference between the `fs` and `gcf` as is is how they check existence of directories.

On the other hand we should not use exist (using exist before writing creates a possible race condition) but I am afraid we will have to use it to check to distinguish between if one should use PUT or POST on aliases.